File: CharacterCounting/CharacterCounting.java
import java.io.*;
import java.util.Scanner;

public class LeeD_Project2_Java{
	
	static char charIn;
	static int index;
	static int charCounts[] = new int[256];
	static Scanner inFile;
	static File file;
	static PrintWriter outFile;
	
	public static void printAry() {
		index = 0;
		for(int i=0; i<256; i++) {
			char symbol = (char) index;
				if(charCounts[index]>0) {
					String charLine ="";
					charLine+= symbol;
					charLine+= " # ";
					charLine+= charCounts[index];
					outFile.println(charLine);
			}
			else {
				//do nothing
			}
			index++;
		}
	}
	
	public static void main(String []args) throws FileNotFoundException{
		inFile = new Scanner(new FileReader(args[0]));
		file = new File(args[1]);
		outFile = new PrintWriter(file);
		
		while(inFile.hasNext()) {
			String token = inFile.next();
			for(int i=0; i<token.length();i++) {
				charIn = token.charAt(i);
				index = (int)charIn;			
				charCounts[index]++;
			}
		}//while until reaches end of data
		
		printAry();
		
		inFile.close();
		outFile.close();
	}//main constructor
	
}//Project class

File: CharacterCounting/CharacterCounting.cpp
#include <string>
#include <iostream>
#include <fstream>
#include <sstream>
using namespace std;

string printAry(int charArray[]){
     stringstream ss;
     int index = 0;
     string charLine;
     while(index < 256){
          if(charArray[index]>0){
               char symbol = (char) index;
               ss<<symbol;
               ss<<" # ";
               ss<<charArray[index];
               ss<<'\n';
               charLine = ss.str();
          }
          else{
               //do nothing
          }
          index++;
     }
     return charLine;
}

int main(int argc, char **argv){
     char charIn;
     int index = 0 ;
     int charCounts[256] = {0};
     ifstream inFile;
     ofstream outFile;
     inFile.open(argv[1]);
     outFile.open(argv[2]);
     while(!inFile.eof()){
          inFile>>charIn;
          index = (int)charIn;
          charCounts[index]++;
     }
     outFile<<printAry(charCounts);
     inFile.close();
     outFile.close();
     return 0;
}
